ops/relation.py

```diff
@@ -0,0 +1,548 @@
+# Copyright 2021 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import semantic_version as semver
+
+from ops.charm import CharmEvents
+from ops.framework import (
+    Object, EventSource, EventBase, StoredState
+)
+
+logger = logging.getLogger(__name__)
+
+
+class Provider(Object):
+    """Manages relations of resource providers
+
+    A :class:`Provider` object manages relations of a resource provider
+    charms, with charms that consume those resources. When a consumer
```

Comment:
Do we have a name for this that isn't 'resource'?
I'm concerned about overloading the term when we have things that are resources (resource-get)

Reply:
How about "service" ?

Reply:
"Resource" has been changed to "service".